File: kubik/kubik-server/src/main/java/com/cspinformatique/kubik/server/domain/product/service/ProductDoublesService.java
package com.cspinformatique.kubik.server.domain.product.service;

import java.util.List;

import com.cspinformatique.kubik.server.model.product.ProductDoubles;

public interface ProductDoublesService {
	List<ProductDoubles> findAll();

	ProductDoubles findOne(String ean13);
}

File: kubik/kubik-server/src/main/webapp/public/js/product/product-doubles/product-doubles-service.js
(function(){
	var PRODUCT_DOUBLES_URL = "/product-doubles";
	
	angular
		.module("Kubik")
		.factory("productDoublesService", ProductDoublesServices);
	
	function ProductDoublesServices($http){
		return {
			findAll : findAll,
			findOne : findOne
		};
		
		function findAll(){
			return $http
				.get(PRODUCT_DOUBLES_URL)
				.then(getSuccess);
			
			function getSuccess(response){
				return response.data;
			}
		}
		
		function findOne(ean13){
			return $http
				.get(PRODUCT_DOUBLES_URL + "/" + ean13)
				.then(getSuccess);
				
			function getSuccess(response){
				return response.data;
			}
		}
	}
})();